File: 2015/Day3/day3_part2.py
inputFile = open("input.txt", 'r')
x, y = 0, 0
positions = {(0, 0)}

inData = inputFile.read()

for i, ch in enumerate(inData):
    if i % 2 == 0:
        if ch == '>':
            x += 1
        elif ch == '<':
            x -= 1
        elif ch == '^':
            y += 1
        elif ch == 'v':
            y -= 1

    positions.add((x,y))

x, y = 0, 0

for i, ch in enumerate(inData):
    if i % 2 != 0:
        if ch == '>':
            x += 1
        elif ch == '<':
            x -= 1
        elif ch == '^':
            y += 1
        elif ch == 'v':
            y -= 1

    positions.add((x,y))

print(len(positions))

File: 2015/Day3/day3_Part2.cpp
#include <iostream>
#include <fstream>
#include <set>
#include <string>

using namespace std;

int main()
{
    ifstream inData;
    set<string> positions;
    int  count = 0, xs = 0, ys = 0, xr = 0, yr = 0, i = 0;
    char ch;
    
    string position = "00";
    positions.insert(position);

    inData.open("input.txt");
    while (inData)
    {
        inData >> ch;

        if (i % 2 == 0)
        {
            if (ch == '>')
                xs++;
            else if (ch == '<')
                xs--;
            else if (ch == '^')
                ys++;
            else if (ch == 'v')
                ys--;
            else
                continue;

            position[0] = xs;
            position[1] = ys;
            positions.insert(position);
        }

        if (i % 2 != 0)
        {
            if (ch == '>')
                xr++;
            else if (ch == '<')
                xr--;
            else if (ch == '^')
                yr++;
            else if (ch == 'v')
                yr--;
            else
                continue;

            position[0] = xr;
            position[1] = yr;
            positions.insert(position);
        }

        ch = ' ';
        i++;
    }
    
    inData.close();
    
    cout << positions.size() - 1;
}